lib/config/generate.js

```diff
@@ -0,0 +1,217 @@
+var path = require('path');
+var Vow = require('vow');
+var vowNode = require('vow-node');
+var Table = require('cli-table');
+var prompt = require('prompt');
+var fs = require('fs');
+
+var Checker = require('../checker');
+var utils = require('../utils');
+
+prompt.message = '';
+prompt.delimiter = '';
+prompt.start();
+
+var get = vowNode.promisify(prompt.get.bind(prompt));
+
+/**
+ * Script that walks the user through the autoconfig flow
+ * @type {String[]}
+ */
+var prompts = [
+    {
+        name: 'Please choose a preset number:'.green,
+        require: true,
+        pattern: /[0-9]*/
+    },
+    {
+        name: 'Create an (e)xception for this rule, or (f)ix the errors yourself?'.green,
+        require: true,
+        pattern: /(e|f)/
+    }
+];
+
+/**
+ * Generated configuration through the autoconfig flow
+ * @type {Object}
+ */
+var config = {};
+
+/**
+ * Generates a configuration object based for the given path
+ * based on the best fitting preset
+ *
+ * @param  {String} path - The path containing file(s) used to guide the configuration
+ *
+ * @return {Promise} Resolved with the generated, JSCS configuration
+ */
+function generateConfig(path) {
```

Comment:
Agreed about the clarity. Forcing an instantiation for what should be a function didn't seem correct. This function only looks like a class because the tests need access to internal methods for stubbing. 

I'll convert this to a class though; thanks for reaffirming.

Tests ruin everything :)